### PR TITLE
feat: added permissions page filter

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/tasks/15763226

## Description

- Added filters to permissions page
- Added permissions and agreements pages to be visible on prod

Resolves [1824](https://github.com/JoinColony/colonyCDapp/issues/1824), [1908](https://github.com/JoinColony/colonyCDapp/issues/1908), [1920](https://github.com/JoinColony/colonyCDapp/issues/1920)

### CODE REVIEW

File: src/components/frame/Extensions/pages/PermissionsPage/partials/PermissionsPageFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+                  onChange({
+                    ...filterValue,
+                    [value]: !filterValue[value],
+                  });
+                }}
+                classNames="subnav-button px-0 sm:px-3.5"
+              >
+                {checkboxLabel}
+              </Checkbox>
+              <>
```
Comment (ArmandoGraterol) [inline src/components/frame/Extensions/pages/PermissionsPage/partials/PermissionsPageFilter.tsx:81]: This empty bracket is not needed

File: src/components/frame/Extensions/pages/PermissionsPage/partials/PermissionsPageFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+              tooltipProps={getTooltipProps}
+              withTooltipStyles={false}
+              cardProps={{
+                rounded: 's',
+                hasShadow: true,
+                className: 'py-6 px-2',
+              }}
+              classNames="w-full sm:max-w-[13.25rem] mr-2"
+            >
+              <>
```
Comment (ArmandoGraterol) [inline src/components/frame/Extensions/pages/PermissionsPage/partials/PermissionsPageFilter.tsx:142]: Same here

File: src/components/frame/Extensions/pages/PermissionsPage/partials/PermissionsPageFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+                          withTooltipStyles={false}
+                          cardProps={{
+                            rounded: 's',
+                            hasShadow: true,
+                            className: 'py-6 px-2',
+                          }}
+                          classNames="w-full sm:max-w-[13.25rem] mr-2"
+                        >
+                          <>
+                            <span className="text-4 text-gray-400 px-3.5 uppercase">
```
Comment (ArmandoGraterol) [inline src/components/frame/Extensions/pages/PermissionsPage/partials/PermissionsPageFilter.tsx:174]: Same here

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
I've noticed this warning appears every time I open the permission types popover:
Screenshot 2024-02-15 at 13 47 19

**Review by warrencolliver — DISMISSED** [review]
all looks good to me! 

filter checklist & search filter work fine and the empty permission types are hidden.

**Review by ArmandoGraterol — DISMISSED** [review]
Nice work! Thank you for addressing the requested changes :)

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
This is looking really good so far, but I've noticed a few issues.

The main issue is that users with custom permissions do not show at all on the members page.

I assume the Reputation Weighted extension should not show if it is not installed / enabled? May need @arrenv to confirm.

When just the "custom" filter type is selected, there is no active filter shown.

https://github.com/JoinColony/colonyCDapp/assets/34915414/381abf96-d6ab-48fc-8a8e-1289fe457c42

On mobile, when opening the filters modal, I get the `Each child in a list should have a unique "key" prop.` error in the console.

The entire page shifts when opening the user popover.

https://github.com/JoinColony/colonyCDapp/assets/34915414/9296323c-58ef-4a8f-954a-dd8a1a445bea

**Comment by arrenv** [comment]
> I assume the Reputation Weighted extension should not show if it is not installed / enabled? May need @arrenv to confirm.

Basically, if it has permissions then it should show on the permissions page. If it does not have permissions, which is likely if not installed, or possible if not enabled, then correct, it should not appear.

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl great start on this! Some feedback points:

1. Icon should be 18x18px as per the design system component - 
image
 
 https://www.figma.com/file/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?type=design&node-id=1782-13112&mode=design&t=WFM99rbhmm6YoSHO-4
 
 2. The two buttons next to each other have different heights. They are both a secondary button and should be the same. Check the design system link in Figma above to check both buttons to sure they are correct. 
 
image

3. The active pill design should match the one @joanna-pagepro has in her activity feed PR #1914 since this is the correct stysling (x icon and padding are incorrect)
image

4. Search input has the incorrect placeholder copy in the input
image

5. There are alignment problems inside the filter pill on mobile when a filter is active
image

6. Mobile needs to be updated with the correct copy in both places
image

7. The size of the X inside the input when content is added doesn't match Figma on desktop and mobile.
image

**Comment by CzarekDryl** [comment]
@arrenv and @iamsamgibbs Issue with showing not installed extensions will be handled in other task - [1872](https://github.com/JoinColony/colonyCDapp/issues/1872)

**Review by iamsamgibbs — DISMISSED** [review]
This is looking better, thanks for addressing those UI issues!

**Review by melyndav — DISMISSED** [review]
Hey @CzarekDryl thanks again for your continued work on the PR.

1. There seems to be an issue with the counting. Here you can see I have 4 filters active in the UI but it's showing as 10 (second screenshot).

It's because when you de-select the custom option, it doesn't de-select all of the sub options under custom and still counts them. But I can only see 4 (on mobile) unless I re-select the custom option to view the others.

Please implement: If custom is deselected in the top level at any point, all child options should deselect as well. Thanks! 

image

image

Desktop 
image


2. Chevron arrow on mobile is the wrong colour token and should be the gray 900.

3. Can we get the correct filter & sort heading added in?

image

4. For searching empty states, can we match the styling of every other no search result component with the border and padding? All the text etc should remain the same, but just make it consistent with the other filters you're implementing for desktop and mobile.

image

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Thanks for the changes, the requests Mel requested are looking all resolved except for the mobile filtering showing an incorrect count.

https://github.com/JoinColony/colonyCDapp/assets/34915414/5456fd7e-b3e3-439b-a646-c780d5e8b0c9

**Review by iamsamgibbs — APPROVED** [review]
Perfect, thanks for the fix!

**Review by ArmandoGraterol — APPROVED** [review]
Thank you for the changes! Looks good to me 👌

**Review by rdig — APPROVED** [review]
Approving so we can get this merged

**Pull request merged**